server/lib/activity-event.js

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var _ = require('lodash');
+var Promise = require('bluebird');
+
+var ALLOWED_QUERY_PARAMS = [
+  'context',
+  'entrypoint',
+  'migration',
+  'service',
+  'utm_campaign',
+  'utm_content',
+  'utm_medium',
+  'utm_source',
+  'utm_term'
+];
+var MAX_PARAM_LENGTH = 100;
+
+module.exports = function (event, data, request) {
+  var queryParams = _.pick(request.query, ALLOWED_QUERY_PARAMS);
+  var eventData = _.assign({
```

Comment:
I didn't know about `_.extend` myself so I just went and looked them up. `_.assign` only copies own properties whereas `_.extend` also copies from the prototype chain.

So in this particular case, the behaviour is the same. I'm happy to switch to `_.extend` if that is the prevalent choice elsewhere in this repo.